derive_miniconf/src/lib.rs

```diff
@@ -218,45 +201,61 @@ fn derive_struct(mut typedef: TypeDefinition, data: syn::DataStruct) -> TokenStr
                 value: &mut [u8]
             ) -> Result<usize, miniconf::Error> {
                 let field = path_parts.next().ok_or(miniconf::Error::PathTooShort)?;
+                let peek = path_parts.peek().is_some();
 
                 match field {
                     #(#get_path_arms ,)*
                     _ => Err(miniconf::Error::PathNotFound)
                 }
             }
 
+            fn next_path<const TS: usize>(
+                state: &mut [usize],
+                path: &mut miniconf::heapless::String<TS>
+            ) -> Result<bool, miniconf::IterError> {
+                let original_length = path.len();
+                loop {
+                    match *state.first().ok_or(miniconf::IterError::PathDepth)? {
+                        #(#next_path_arms ,)*
+                        _ => return Ok(false),
+                    };
+
+                    // Note(unreachable) Without any deferred fields, every arm above returns
+                    #[allow(unreachable_code)]
```

Comment:
Can (should?) we check for at least 1 deferred field before calling into this? Not sure if it's worth it though.

Reply:
Since it's just dead code even in the case where the compiler notices it (it does) but doesn't optimize it away, I think this is minor.